doc/developer/platform/ux.md

```diff
@@ -1,20 +1,945 @@
+choose when upgrades happen, nor will they be permitted to opt out of upgrades.
+
+Initially, version upgrades require downtime and so will be scheduled during a
+maintenance window. Our goal is to provide zero downtime upgrades, at which
+point we will no longer need maintenance windows.
+
+For a discussion of this policy, see [upgrade cadence](#upgrade-cadence) below.
+
+### Observability
+
+We have two separate challenges ahead of us: internal observability and
+external observability.
+
+Internal observability is about providing our customer support engineers and
+production engineers with the ability to diagnose problems in our
+infrastructure. Materialize Platform will be more difficult to observe than
+`materialized` today, because distributed systems are more challenging to
+monitor than single-node systems. The root cause of an issue may be a machine
+running a different service, and tracing causal chains across machines and
+service boundaries is vastly more difficult than tracing function calls through
+a single process. Work items revolve around building distributed tracing and
+monitoring infrastructure using technologies like the [`tracing`] crate,
+[Prometheus], and [OpenTelemetry]. This work has already started.
+
+TODO: engage @guswynn and @antifuchs to think further about internal
+observability.
+
+External observability is about providing our customers with insight into the
+performance of their Materialize Platform account. We can't provide our
+customers with our distributed tracing infrastructure, as that will leak private
+details of our infrastructure and is a poor user experience to boot. Instead,
+we'll want to provide structured access to curated diagnostic information via
+the system catalog. Snowflake's [`QUERY_HISTORY`] view is a good example of this
+approach.
+
+A key implication is that there will be no log file in Materialize Platform.
+What processes' logs would we even display? Details about what processes are
+running will be hidden from users. Any log messages in `materialized` today that
+are meant for end-users will need to instead make their messages available via
+tables or sources in the system catalog.
+
+## Discussion
+
+What follows are my (@benesch's) thoughts on the design tradeoffs that were
+evaluated in the above proposal.
+
+### Why are regions siloed?
+
+The ideal user experience would do away with the region silo. Let clusters
+access sources from any region. Let clusters span regions and failover between
+regions.
+
+But sharing infrastructure across regions causes high latency, high network
+transfer fees, and introduces hard distributed systems problems. We don't have a
+good handle on how to solve those problems. Colocating the important
+infrastructure in a single region eliminates all these problems.
+
+Put another way: we are not confident we can deliver a cross-region Materialize
+Platform, but we are confident we can deliver a single-region Materialize
+Platform.
```

Comment:
I'd imagine the 5/10 year plan would involve multi-region, yeah?

Reply:
Probably, yeah, but there are different degrees of support for multi-region. One relatively easy first step is to allow sources to be read from multiple regions (at the cost of network transfer fees). The next step might be allowing compute clusters to have replicas in multiple regions, but not offering linearizability across regions. I'm not sure we ever get to the point where we support linearizability across regions, because the performance implications are grim.

Reply:
The intent here, though, is to give ourselves the freedom to entirely ignore multi-region concerns for now! You're right that we'll probably need to bolt on some multi-region support in the future, and it will always feel like a bolted-on feature, a la [Global Snowflake](https://docs.snowflake.com/en/user-guide/secure-data-sharing-across-regions-plaforms.html) or [S3 Multi-Region](https://aws.amazon.com/s3/features/multi-region-access-points/). But we can entirely kick that can down the road for now.